src/components/ContextMenu/index.js

```diff
@@ -0,0 +1,386 @@
+import $ from "jquery";
+import cx from "classnames";
+import React from "react";
+import { compose, withStateHandlers, withProps, lifecycle } from "recompose";
+import { MenuItem } from "react-bootstrap";
+import { i18n } from "../../js/i18n";
+import DropdownMenu from "./DropdownMenu";
+import InputHelperModal from "./InputHelperModal";
+import LiveHelperModal from "./LiveHelperModal";
+import PrefModal from "./PrefModal";
+
+function noop() {}
+
+const EVENT_KEY_BY_HOT_KEY = {
+  ["C".charCodeAt(0)]: "copy",
+  ["E".charCodeAt(0)]: "copyLinkUrl",
+  ["P".charCodeAt(0)]: "paste",
+  ["S".charCodeAt(0)]: "searchGoogle",
+  ["T".charCodeAt(0)]: "openUrlNewTab"
+};
+
+const menuHandlerByEventKey = {
+  copy: (pttchrome, { selectedText }) => pttchrome.doCopy(selectedText),
+  copyAnsi: pttchrome => pttchrome.doCopyAnsi(),
+  paste: pttchrome => pttchrome.doPaste(),
+  searchGoogle: (pttchrome, { selectedText }) =>
+    pttchrome.doSearchGoogle(selectedText),
+  openUrlNewTab: (pttchrome, { aElement }) =>
+    pttchrome.doOpenUrlNewTab(aElement),
+  copyLinkUrl: (pttchrome, { contextOnUrl }) => pttchrome.doCopy(contextOnUrl),
+  selectAll: pttchrome => pttchrome.doSelectAll(),
+  mouseBrowsing: pttchrome => pttchrome.switchMouseBrowsing()
+};
+
+const onPrefSaveImpl = (pttchrome, values) => {
+  pttchrome.onValuesPrefChange(values);
+  pttchrome.modalShown = false;
+  pttchrome.setInputAreaFocus();
+  pttchrome.switchToEasyReadingMode(pttchrome.view.useEasyReadingMode);
+
+  return {
+    openSettings: false
+  };
+};
+
+const initialState = {
+  // --- Menu state ---
+  open: false,
+  pageX: 0,
+  pageY: 0,
+  contextOnUrl: "",
+  aElement: undefined,
+  selectedText: "",
+  urlEnabled: false,
+  normalEnabled: false,
+  selEnabled: false,
+  // --- Modal state ---
```

Comment:
I feel like this should be states in `App` rather than in `ContextMenu`. And those `HelperModal`s should be independent UI modules, in parallel with `ContextMenu`. But let me know if there's some difficulties of doing this.

Reply:
> I feel like this should be states in App rather than in ContextMenu.

Agreed. But I'd rather do it in the future refactorings that refactors `App` to React components.

> And those HelperModals should be independent UI modules, in parallel with ContextMenu

Disagreed. Because these `HelperModals` are only shown by the `ContextMenu`, which makes perfect sense to hide them under the folder of the `ContextMenu` to reduce cognitive overhead.

Reply:
I'm surprised that you agree former but not latter. I'm okay to keep them as now if there's too much work pulling those helper modals out.

Reply:
I believe that component tree should be aligned with file/directory tree.